### ISSUE TITLE
[JIT-autodiff] batch_norm isn't differentiable

### ISSUE THREAD

**Issue by arthuryuan1987** [pytorch/pytorch#52589]
## Motivation

Hi, @Krovatkin ,

We have a use case of JIT training on ResNet50 to make fusion work both in forward and backward path. So far, our fusable pattern is BN_FWD + ReLU and ReLU + BN_BWD.

But batch_norm is disabled in `autodiff`. 
https://github.com/pytorch/pytorch/blob/9529ae377629f4253af0720cb01f09db6fcad4c9/torch/csrc/jit/runtime/symbolic_script.cpp#L992.

**Comment by arthuryuan1987** [comment]
@gujinghui @jgong5

**Comment by Krovatkin** [comment]
@jjsjann123  Did you guys also hit this?

**Comment by Krovatkin** [comment]
@arthuryuan1987 I'll take a look

**Comment by jjsjann123** [comment]
Yeah, I briefly mentioned this in our last meeting and I'm also working on a solution there.

Our codegen doesn't support inplace operations yet, I'm still trying to figure out how to get it to an upstreamable state.

**Comment by arthuryuan1987** [comment]
Hi, @jjsjann123 & @Krovatkin ,

Could you still take inplace `relu_` into account? Our fusion pattern is `bn` + `relu_` on ResNet50. We prefer not modifying ResNet50 model.
So if both `bn` & `relu_` are differentiable, they can be merged to one differetiable graph.
Or have you any other idea of backend custom fusion optimization, like `bn` + `relu_`?

**Comment by arthuryuan1987** [comment]
@majing921201

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
